### ISSUE TITLE
Comma dangle syntax no longer ignores functions: never

### ISSUE THREAD

**Issue by willlma** [eslint/eslint#13165]
**Tell us about your environment**

* **ESLint Version:**
Upgraded from 6.51 to 6.8.0
* **Node Version:**
v13.12.0
* **npm Version:**
Using yarn: 1.22.4

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
extends: airbnb
parser: babel-eslint
env:
  browser: true
rules:
  arrow-parens: [warn, always]
  comma-dangle: [warn, always-multiline, { "functions": never }]
  import/prefer-default-export: 0
  jsx-quotes: [warn, prefer-single]
  max-len: [warn, { code: 80 }]
  object-curly-newline: ["warn", { "multiline": true, "consistent": true }]
  operator-linebreak: [warn, after]
  radix: [warn, as-needed]
  react/destructuring-assignment: [warn, always, { ignoreClassFields: true }]
  react/jsx-filename-extension: 0
  react/jsx-props-no-spreading: 0
  react/state-in-constructor: 0
settings:
  import/resolver:
    webpack:
      config: './config/webpack/custom.js'

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const store = createStore(
  combineReducers({ ...reducers }),
  initialState,
  applyMiddleware(thunk)
);
```

```bash
yarn lint
```

**What did you expect to happen?**
No warnings. My `.eslintrc` specifies
```
comma-dangle: [warn, always-multiline, { "functions": never }]
```
which worked prior to this upgrade

**What actually happened? Please include the actual, raw output from ESLint.**
 ```
warning  Missing trailing comma  comma-dangle
```

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by mdjermanovic** [comment]
Hi @willlma, thanks for the bug report!

The actual bug, in this case, is that the rule doesn't report invalid configuration and just ignores the extra option `{ "functions": never }`.

Your configuration works the same as:

```yaml
comma-dangle: [warn, always-multiline]
```

Per the [documentation](https://eslint.org/docs/rules/comma-dangle#options), it should be either a string option or an object option. This particular rule doesn't support overriding the specified string option with an additional object option.

The correct configuration would be:

```yaml
comma-dangle: [warn, {
    "arrays": "always-multiline",
    "objects": "always-multiline",
    "imports": "always-multiline",
    "exports": "always-multiline",
    "functions": "never" 
}]
```

In v6.5.1 `[warn, always-multiline]` actually used to ignore functions (it didn't disallow comma dangle), which was fixed in #12462.

**Comment by mdjermanovic** [comment]
Reopening because this looks like a bug in the rule's schema.

**Comment by kaicataldo** [comment]
@mdjermanovic Do you mind marking this as accepted if you've verified this bug and outlining what you believe the correct fix is here? That way someone from the community could pick this up!

**Comment by mdjermanovic** [comment]
Marked as accepted. PR #13166 to fix this is prepared, we're just not sure whether this can be a semver-minor fix or should we treat it as a breaking change.
